libraries/chain/include/graphene/chain/protocol/fee_schedule.hpp

```diff
@@ -34,6 +34,49 @@ namespace graphene { namespace chain {
    };
    typedef transform_to_fee_parameters<operation>::type fee_parameters;
 
+   template<typename Operation>
+   class fee_helper {
```

Comment:
unsure about why the fee_helper was needed to be implemented, sure it haves a reason but can you explain a bit further this and about why the `account_create_operation` and the new created `bid_collateral_operation` are special ?

Reply:
The challenge here was to create a fee that defaults to the value of another fee, as opposed to a static default. For this I needed to specialize the function template, and trying to do that I ran into the trap that partial specialization of certain templates is not allowed. The solution was to introduce a new class fee_helper that can be specialized as needed.

Reply:
account_create is special because it is the only operation that uses the non-const get().
bid_collateral is special because the fee defaults to the current value of call_order_update.

Reply:
got it, thanks!